UefiPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -286,3 +288,39 @@ PlatformBootManagerUnableToBoot (
   return;
 }
 
+
+
+/**
+  Get/update PcdBootManagerMenuFile from GUID HOB which will be assigned in bootloader.
+
+  @retval EFI_SUCCESS       The entry point is executed successfully.
+  @retval other             Some error occurs.
+  
+**/
+EFI_STATUS
+EFIAPI
+PlatformBootManagerLibConstructor (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  ) 
+{
+  EFI_STATUS                         Status;
+  UINTN                              size;
+  VOID                               *GuidHob;
+  BOOT_MANAGER_MENU_FILE             *BootManagerMenuFile;
+  Status = EFI_SUCCESS;
+  GuidHob = GetFirstGuidHob (&gUniversalPayloadBootManagerMenuFileGuid);
+  DEBUG((DEBUG_ERROR,"gUniversalPayloadBootManagerMenuFileGuid\n"));
+  //
+  // Find the buffer information and update PCDs
+  //
+  if (GuidHob != NULL) {
+    DEBUG((DEBUG_ERROR,"gUniversalPayloadBootManagerMenuFileGuid2\n"));
+    BootManagerMenuFile = (BOOT_MANAGER_MENU_FILE *)GET_GUID_HOB_DATA (GuidHob);
+    size = sizeof(BootManagerMenuFile->PassedGUID);
+    Status = PcdSetPtrS (PcdBootManagerMenuFile, &size, &BootManagerMenuFile->PassedGUID);

```

Comment:
1. Check AutoGen.c/h to make sure the PcdSetPtrS() macro is expanded as expected.
2. size should be Size.
Did you run ECC to check coding style?